src/v/pandaproxy/schema_registry/seq_writer.h

```diff
@@ -85,25 +86,37 @@ class seq_writer final : public ss::peering_sharded_service<seq_writer> {
               });
         };
 
-        return container().invoke_on(0, _smp_opts, remote);
+        return container().invoke_on(0, _smp_opts, remote).then([](auto res) {
+            return std::move(res).value();
+        });
     }
 
     /// The part of sequenced_write that runs on shard zero
     ///
     /// This is declared as a separate member function rather than
     /// inline in sequenced_write, to avoid compiler issues (and resulting
     /// crashes) seen when passing in a coroutine lambda.
-    template<typename F>
-    ss::future<typename std::invoke_result_t<F, model::offset, seq_writer&>::
-                 value_type::value_type>
+    ///
+    /// The return of f is wrapped in an outcome<>, to transport exceptions
+    /// without causing a retry.
+    template<
+      typename F,
+      typename invoke_result_t = typename std::
+        invoke_result_t<F, model::offset, seq_writer&>::value_type::value_type>
+    ss::future<
+      outcome::outcome<invoke_result_t, std::error_code, std::exception_ptr>>
     sequenced_write_inner(F f) {
         // If we run concurrently with them, redundant replays to the store
         // will be safely dropped based on offset.
         co_await read_sync();
 
         auto next_offset = _loaded_offset + model::offset{1};
-        auto r = co_await f(next_offset, *this);
-
+        std::optional<invoke_result_t> r;
+        try {
```

Comment:
Should this try block extend up to also include `read_sync()`, in case that throws an exception?

Reply:
> Should this try block extend up to also include `read_sync()`, in case that throws an exception?

I'm not sure. I feel like if `retry_sync` throws, it should be retried.

Reply:
I was thinking the kafka client might throw on both read+write paths during shutdown perhaps?  But I'm speculating - your call.

Reply:
The client outlives the listener, so that case shouldn't happen. There may be other cases. I feel like it's all I/O, though, and retries are reasonable.